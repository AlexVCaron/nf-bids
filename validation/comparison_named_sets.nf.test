nextflow_workflow {

    name "Plugin Comparison: Named Sets"
    script "./main.nf"
    workflow "main_workflow"

    test("qmri_mtsat - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_mtsat"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_mtsat.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_tb1tfl - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_tb1tfl"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_tb1tfl.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }
}
