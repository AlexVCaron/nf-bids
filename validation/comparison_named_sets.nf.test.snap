{
    "qmri_mtsat - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"MTS\":{\"PDw\":{\"json\":\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-off_MTS.json\",\"nii\":\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-off_MTS.nii.gz\"},\"MTw\":{\"json\":\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-on_MTS.json\",\"nii\":\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-on_MTS.nii.gz\"},\"T1w\":{\"json\":\"qmri_mtsat/sub-01/anat/sub-01_flip-2_mt-off_MTS.json\",\"nii\":\"qmri_mtsat/sub-01/anat/sub-01_flip-2_mt-off_MTS.nii.gz\"}},\"TB1DAM\":{\"json\":[\"qmri_mtsat/sub-01/fmap/sub-01_flip-1_TB1DAM.json\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-2_TB1DAM.json\"],\"nii\":[\"qmri_mtsat/sub-01/fmap/sub-01_flip-1_TB1DAM.nii.gz\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-2_TB1DAM.nii.gz\"]}},\"filePaths\":[\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-off_MTS.json\",\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-off_MTS.nii.gz\",\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-on_MTS.json\",\"qmri_mtsat/sub-01/anat/sub-01_flip-1_mt-on_MTS.nii.gz\",\"qmri_mtsat/sub-01/anat/sub-01_flip-2_mt-off_MTS.json\",\"qmri_mtsat/sub-01/anat/sub-01_flip-2_mt-off_MTS.nii.gz\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-1_TB1DAM.json\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-1_TB1DAM.nii.gz\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-2_TB1DAM.json\",\"qmri_mtsat/sub-01/fmap/sub-01_flip-2_TB1DAM.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:03.759498167"
    },
    "qmri_tb1tfl - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"TB1TFL\":{\"anat\":{\"json\":\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-anat_TB1TFL.json\",\"nii\":\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-anat_TB1TFL.nii.gz\"},\"famp\":{\"json\":\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-famp_TB1TFL.json\",\"nii\":\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-famp_TB1TFL.nii.gz\"}}},\"filePaths\":[\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-anat_TB1TFL.json\",\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-anat_TB1TFL.nii.gz\",\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-famp_TB1TFL.json\",\"qmri_tb1tfl/sub-01/fmap/sub-01_acq-famp_TB1TFL.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:08.547573951"
    }
}